src/Lean/Server/Requests.lean

```diff
@@ -150,9 +150,35 @@ def registerLspRequestHandler (method : String)
 
   requestHandlers.modify fun rhs => rhs.insert method { fileSource, handle }
 
-private def lookupLspRequestHandler (method : String) : IO (Option RequestHandler) := do
+def lookupLspRequestHandler (method : String) : IO (Option RequestHandler) := do
   (← requestHandlers.get).find? method
 
+/-- NB: This method may only be called in `builtin_initialize` blocks.
+
+Register another handler to invoke after the last one registered for a method.
+At least one handler for the method must have already been registered to perform
+chaining.
+
+For more details on the registration of a handler, see `registerLspRequestHandler`. -/
+def chainLspRequestHandler (method : String)
+    paramType [FromJson paramType]
+    respType [FromJson respType] [ToJson respType]
+    (handler : paramType → RequestTask respType → RequestM (RequestTask respType)) : IO Unit := do
+  if !(← IO.initializing) then
+    throw <| IO.userError s!"Failed to chain LSP request handler for '{method}': only possible during initialization"
+  if let some oldHandler ← lookupLspRequestHandler method then
+    let handle := fun j => do
+      let t ← oldHandler.handle j
+      let t ← t.map fun x => x.bind fun j => FromJson.fromJson? j |>.mapError fun e =>
+        IO.userError s!"failed to parse original LSP request handler for `{method}` when chaining: {e}"
```

Comment:
```suggestion
        IO.userError s!"Failed to parse original LSP response for `{method}` when chaining: {e}"
```
Since I guess it's not the handler being parsed :)